File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        boolean answer = false;
        Stack<Character> stack = new Stack<>();
        
        for (char ch : s.toCharArray()) {
            if (ch == '(') {
                stack.push(ch);
            } else if (!stack.empty() && ch == ')') {
                stack.pop();
            } else if (stack.empty() && ch == ')') {
                return false;
            }
        }
        
        if (stack.empty()) 
            answer = true;

        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = False
    stack = []
    
    for str in s:
        if str == "(":
            stack.append(str)
        elif len(stack) > 0 and str == ")":
            stack.pop()
        elif len(stack) == 0 and str == ")":
            return False
    
    if len(stack) == 0:
        answer = True
    
    
    return answer